gen3/bin/replicate.sh

```diff
@@ -0,0 +1,205 @@
+    fi
+    local destinationAccountId=$(gen3_aws_run aws iam get-role --role-name batch-operations-role --profile "$profile" | jq -r .Role.Arn | cut -d : -f 5)
+    local destinationRolePolicy="{\"Version\": \"2012-10-17\",\"Statement\": [{\"Sid\": \"AllowBatchOperationsDestinationObjectCOPY\",\"Effect\": \"Allow\",\"Action\": [\"s3:PutObject\",\"s3:PutObjectVersionAcl\",\"s3:PutObjectAcl\",\"s3:PutObjectVersionTagging\",\"s3:PutObjectTagging\",\"s3:GetObject\",\"s3:GetObjectVersion\",\"s3:GetObjectAcl\",\"s3:GetObjectTagging\",\"s3:GetObjectVersionAcl\",\"s3:GetObjectVersionTagging\"],\"Resource\": [\"arn:aws:s3:::$source/*\",\"arn:aws:s3:::$destinaion/*\"]}]}"
+# need to only make a joint policy if there was a policy previously.
+    gen3_log_info "Checking for old policies and modfying bucket policy to add batch operations policy"
+    resetPolicy=$(gen3_aws_run aws s3api get-bucket-policy --bucket $source | jq -r .Policy )
+    local oldPolicy=$(gen3_aws_run aws s3api get-bucket-policy --bucket $source | jq -r .Policy | jq -r .Statement[] )
+    local newPolicy=$(echo "{\"Sid\": \"AllowBatchOperationsSourceManfiestRead\",\"Effect\": \"Allow\",\"Principal\": {\"AWS\": [\"arn:aws:iam::$destinationAccountId:role/batch-operations-role\"]},\"Action\": [\"s3:GetObject\",\"s3:GetObjectVersion\"],\"Resource\": \"arn:aws:s3:::$source/*\"}" |jq -r .)
+    if [[ -z $oldPolicy ]]; then
+      local bucketPolicy='{"Version": "2012-10-17", "Statement": ['$newPolicy']}'
+    else
+      gen3_log_info "Found old bucket policy. Modifying for new policy $oldPolicy"
+      local bucketPolicy='{"Version": "2012-10-17", "Statement": ['$oldPolicy','$newPolicy']}'
+    fi
+    # sleep to allow role to fully generate for policy
+    sleep 5
+    gen3_aws_run aws s3api put-bucket-policy --bucket $source --policy "$bucketPolicy"
+    if [[ ! -z $(gen3_aws_run aws iam list-role-policies --role-name batch-operations-role --profile "$profile" | jq -r .PolicyNames[]) ]]; then
+      gen3_log_info "Old policy exists. Removing in favor of new policy"
+      gen3_aws_run aws iam delete-role-policy --role-name batch-operations-role --policy-name batch-operations-policy --profile $profile
+    fi
+    local policy="{\"Version\": \"2012-10-17\", \"Statement\": [ { \"Action\": [ \"s3:PutObject\", \"s3:PutObjectAcl\", \"s3:PutObjectTagging\" ], \"Effect\": \"Allow\", \"Resource\": \"arn:aws:s3:::$destination/*\" }, { \"Action\": [ \"s3:GetObject\", \"s3:GetObjectAcl\", \"s3:GetObjectTagging\" ], \"Effect\": \"Allow\", \"Resource\": \"arn:aws:s3:::$source/*\" }, { \"Effect\": \"Allow\", \"Action\": [ \"s3:GetObject\", \"s3:GetObjectVersion\", \"s3:GetBucketLocation\" ], \"Resource\": [ \"arn:aws:s3:::$destination/*\" ] }, { \"Effect\": \"Allow\", \"Action\": [ \"s3:PutObject\", \"s3:GetBucketLocation\" ], \"Resource\": [ \"arn:aws:s3:::$destination/*\" ] }    ]}"
```

Comment:
Did you test this one too? Looking at the quotes escaping, just want to make sure you don't have to patch this again later.